pwndbg/gdblib/kernel/__init__.py

```diff
@@ -153,11 +188,15 @@ def phys_to_pfn(self, phys: int) -> int:
     def pfn_to_phys(self, pfn: int) -> int:
         return pfn << self.PAGE_SHIFT
 
-    def phys_to_page(self, phys: int) -> int:
-        return (self.STRUCT_PAGE_SIZE * phys_to_pfn(phys)) + self.START_KERNEL_map
+    def pfn_to_page(self, pfn: int) -> int:
+        # assumption: SPARSEMEM_VMEMMAP memory model used
+        # FLATMEM or SPARSEMEM not (yet) implemented
```

Comment:
Is there any way we can detect the memory model?

Reply:
TBH I haven't yet looked into it, that's why I just left it as a comment for now...

Reply:
On first glance it looks like it can be detected entirely from the kernel config and should therefore be straight-forward from the output of the `kconfig()` command. But this might be something for another day ;)

Reply:
I think (tm) that looking at/for the configs `CONFIG_SPARSEMEM`, `CONFIG_SPARSEMEM_EXTREME`, `CONFIG_SPARSEMEM_VMEMMAP` and `CONFIG_FLATMEM` in various combinations might be the key (but there might be other ones that I have overseen)...

Reply:
Shall we add a comment how it could be detected in the future?

Reply:
I added the reference to your link to the `ArchOps` class and included a note that using that reference and the kernel config other memory models can be supported in the future.